monai/utils/sliding_window_inference.py

```diff
@@ -0,0 +1,165 @@
+# Copyright 2020 MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import math
+import numpy as np
+import torch
+from monai.data.transforms import ImageEndPadder
+
+
+def sliding_window_inference(inputs, roi_size, sw_batch_size, predictor, device):
+    """Use SlidingWindow method to execute inference.
+
+    Args:
+        roi_size (list, tuple): the window size to execute SlidingWindow inference.
+        sw_batch_size (int): the batch size to run window slices.
+        predictor: a portion from numpy.lib.arraypad.pad is copied below.
+        device: on which device to execute model inference, cpu or gpu.
+
+    Note:
+        must be channel first, support both 2D and 3D.
+        input data must have batch dim.
+        execute on 1 image/per inference, run a batch of window slices of 1 input image.
+    """
+    num_spatial_dims = len(inputs.shape) - 2
+    assert len(roi_size) == num_spatial_dims, 'roi_size {} does not match input dims.'.format(roi_size)
+
+    # determine image spatial size and batch size
+    # Note: all input images must have the same image size and batch size
```

Comment:
This should be in docstring or assertion.

Reply:
Hi @madil90 ,

Thanks for your suggestion. I kept this explicit warning comment here because it still has "TODO":
Will refactor this part when we support batch_size > 1 in the future.
Thanks.